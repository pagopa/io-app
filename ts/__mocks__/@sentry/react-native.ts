export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
