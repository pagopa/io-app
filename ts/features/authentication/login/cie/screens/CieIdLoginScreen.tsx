import { RouteProp, useRoute } from "@react-navigation/native";
import { AuthenticationParamsList } from "../../../common/navigation/params/AuthenticationParamsList";
import CieIdLoginWebView from "../../../login/cie/components/CieIdLoginWebView";
import { AUTHENTICATION_ROUTES } from "../../../common/navigation/routes";

const CieIdLoginScreen = () => {
  const route =
    useRoute<
      RouteProp<
        AuthenticationParamsList,
        typeof AUTHENTICATION_ROUTES.CIE_ID_LOGIN
      >
    >();

  return <CieIdLoginWebView {...route.params} />;
};

export default CieIdLoginScreen;
