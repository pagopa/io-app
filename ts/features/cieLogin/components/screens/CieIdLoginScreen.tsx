import React from "react";
import { RouteProp, useRoute } from "@react-navigation/native";
import { AuthenticationParamsList } from "../../../../navigation/params/AuthenticationParamsList";
import ROUTES from "../../../../navigation/routes";
import CieIdLoginWebView from "../CieIdLoginWebView";

const CieIdLoginScreen = () => {
  const route =
    useRoute<
      RouteProp<
        AuthenticationParamsList,
        typeof ROUTES.AUTHENTICATION_CIE_ID_LOGIN
      >
    >();

  return <CieIdLoginWebView {...route.params} />;
};

export default CieIdLoginScreen;
