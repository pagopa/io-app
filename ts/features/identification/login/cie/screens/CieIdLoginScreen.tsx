import { RouteProp, useRoute } from "@react-navigation/native";
import { IdentificationParamsList } from "../../../common/navigation/params/IdentificationParamsList";
import CieIdLoginWebView from "../../../login/cie/components/CieIdLoginWebView";
import { IDENTIFICATION_ROUTES } from "../../../common/navigation/routes";

const CieIdLoginScreen = () => {
  const route =
    useRoute<
      RouteProp<
        IdentificationParamsList,
        typeof IDENTIFICATION_ROUTES.CIE_ID_LOGIN
      >
    >();

  return <CieIdLoginWebView {...route.params} />;
};

export default CieIdLoginScreen;
