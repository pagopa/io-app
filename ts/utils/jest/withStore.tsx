import React, { JSXElementConstructor } from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { appReducer } from "../../store/reducers";
import { applicationChangeState } from "../../store/actions/application";

/**
 * A HOC to provide the redux `Context`
 * @param Component the component to wrap
 * @returns The given `Component` wrapped with the redux `Provider`
 */
export function withStore<P extends Record<string, unknown>>(
  Component: JSXElementConstructor<P>
) {
  const globalState = appReducer(undefined, applicationChangeState("active"));
  const store = createStore(appReducer, globalState as any);

  return (props: P) => (
    <Provider store={store}>
      <Component {...props} />
    </Provider>
  );
}
