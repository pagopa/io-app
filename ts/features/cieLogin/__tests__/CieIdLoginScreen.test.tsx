/* eslint-disable @typescript-eslint/ban-ts-comment */
import { render } from "@testing-library/react-native";
import React from "react";
import { Linking } from "react-native";
import CieIdLoginScreen from "../components/screens/CieIdLoginScreen";
import { withStore } from "../../../utils/jest/withStore";

const TestComponent = withStore(CieIdLoginScreen);
const mockUseRoute = jest.fn();

jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useRoute: () => mockUseRoute,
  useNavigation: () => ({
    replace: jest.fn()
  })
}));

describe(CieIdLoginScreen, () => {
  jest
    .spyOn(Linking, "addEventListener")
    // @ts-ignore
    .mockReturnValue({ remove: jest.fn() });

  it("Should match snapshot", () => {
    const component = render(<TestComponent />);
    expect(component).toMatchSnapshot();
  });
});
