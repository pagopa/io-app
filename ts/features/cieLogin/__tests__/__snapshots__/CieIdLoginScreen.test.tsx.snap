// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CieIdLoginScreen Should match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flex": 1,
      "justifyContent": "center",
      "marginHorizontal": 16,
    }
  }
>
  <View
    javaScriptEnabled={true}
    onError={[Function]}
    onHttpError={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    originWhitelist={
      [
        "https://*",
        "iologin://*",
        "http://*",
      ]
    }
    renderLoading={[Function]}
    source={
      {
        "uri": "undefined/login?entityID=xx_servizicie&authLevel=undefined",
      }
    }
    startInLoadingState={true}
    testID="cie-id-webview"
    userAgent="Mozilla/5.0 (iPhone; CPU iPhone OS 14_0_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Mobile/15E148 Safari/604.1"
  />
</RNCSafeAreaView>
`;
